packages/express-wrapper/src/request.ts

```diff
@@ -72,3 +86,32 @@ export const decodeRequestAndEncodeResponse = (
     },
   );
 };
+
+export async function runMiddlewareChain<
+  Input extends {},
+  Chain extends MiddlewareChain,
+>(
+  input: Input,
+  chain: Chain,
+  req: express.Request,
+  res: express.Response,
+): Promise<MiddlewareChainOutput<Input, Chain>> {
+  let result: {} = input;
+  for (const middleware of chain) {
+    const nextValueOrErr: unknown = await new Promise((resolve, reject) => {
+      try {
+        middleware(req, res, resolve);
+      } catch (err) {
+        reject(err);
+      }
+    });
+
+    let newResult = extractOutput(nextValueOrErr);
+    if (!newResult) {
```

Comment:
Can we explicitly check for `undefined` here? To match the value returned in the `else` in `extractOutput`
```suggestion
    if (newResult === undefined) {
```